### PR TITLE
BUGFIX: TNT-1385 Fix Edit button of dashboard with dashboard permissions disabled, TNT-1384 Preserve sharing status when a dashboard is updated

### PR DESCRIPTION
**Pull request by tomassatinsky**
JIRA: TNT-1385

---

Supported PR commands:

| Command                                                 | Description                                                |
| ------------------------------------------------------- | ---------------------------------------------------------- |
| `ok to test`                                            | Re-run standard checks                                     |
| `extended check sonar`                                  | SonarQube tests                                            |
| `extended check plugins`                                | Dashboard plugins tests                                    |
| `extended test - backstop`                              | BackstopJS tests                                           |
| **E2E Cypress tests commands - TIGER**                  |                                                            |
| `extended test - tiger-cypress - isolated <testName>`   | Run isolated tests running against recorded Tiger backend. |
| `extended test - tiger-cypress - record <testName>`     | Create a new recording for isolated Tiger tests.           |
| `extended test - tiger-cypress - integrated <testName>` | Run integrated tests against live backend                  |
| **E2E Cypress tests commands - BEAR**                   |                                                            |
| `extended test - cypress - isolated <testName>`         | Run isolated tests running against recorded Bear backend.  |
| `extended test - cypress - record <testName>`           | Create a new recording for isolated Bear tests.            |

`<testName>` in cypress commands is used to filter specfiles. Example, to run record with BEAR backend

-   Against `dashboard.spec.ts` and `drilling.spec.ts`, execute command `extended test - cypress - record dashboard,drilling`
-   Against all specfiles, execute command `extended test - cypress - record` or `extended test - cypress - record *`

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `extended test - backstop` passes
-   [ ] `extended test - tiger-cypress - record` to record new mapping files (Tiger BE)
-   [ ] `extended test - cypress - record` to record new mapping files (Bear BE)
-   [ ] `extended test - tiger-cypress - isolated` passes
-   [ ] `extended test - cypress - isolated` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: libs/sdk-ui-dashboard/src/presentation/topBar/buttonBar/button/selectors.ts [hunk 1]
```diff
@@ -20,18 +20,68 @@ import {
...
+    selectDashboardEditModeDevRollout,
+    hasEditDashboardPermission,
+    hasEditLockedDashboardPermission,
     selectDashboardLockStatus,
-    (isEditModeEnabled, canEditDashboardPermission, canEditLockedDashboardPermission, isLocked) =>
-        isEditModeEnabled && canEditDashboardPermission && (!isLocked || canEditLockedDashboardPermission),
+    selectIsReadOnly,
+    (isEditModeEnabled, hasEditDashboardPermission, hasEditLockedDashboardPermission, isLocked, isReadOnly) =>
+        isEditModeEnabled &&
+        !isReadOnly &&
```
Comment (tomassatinsky) [inline libs/sdk-ui-dashboard/src/presentation/topBar/buttonBar/button/selectors.ts:82]: Added the read only condition which was previously in the selector but has been forgotten when refactoring it previously

File: libs/sdk-ui-dashboard/src/presentation/topBar/buttonBar/button/selectors.ts [hunk 2]
```diff
@@ -20,18 +20,68 @@ import {
     selectCanEditLockedDashboardPermission,
     selectCanShareDashboardPermission,
     selectCanShareLockedDashboardPermission,
+    selectCanManageWorkspace,
+    selectCanManageAnalyticalDashboard,
+    selectSupportsHierarchicalWorkspacesCapability,
```
Comment (tomassatinsky) [inline libs/sdk-ui-dashboard/src/presentation/topBar/buttonBar/button/selectors.ts:25]: The new selectors below fallback to the original edit button behavior when dashboard permissions are not enabled

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/11723/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/13757/ : SUCCESS in 9m 58s

File: libs/sdk-backend-tiger/src/backend/workspace/dashboards/index.ts [hunk 3]
```diff
@@ -316,7 +316,13 @@ export class TigerWorkspaceDashboards implements IWorkspaceDashboardsService {
...
         });
 
-        return convertDashboard(result.data, filterContext);
+        /**
+         * Getting the dashboard again to get the shareStatus of the dashboard
+         * When NAS-4822 is completed, we can add `metainclude: ["accessInfo"],` to the payload above
+         * and return just `convertDashboard(result.data, filterContext);` below
+         */
+        const { id, type } = result.data.data;
+        return this.getDashboard(idRef(id, type));
```
Comment (tomassatinsky) [inline libs/sdk-backend-tiger/src/backend/workspace/dashboards/index.ts:325]: We had this code here before some time ago during the development but then removed it as we thought it's no longer needed.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/11725/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/13759/ : SUCCESS in 11m 34s

**Review by xMort — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR3519

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/367/ : FAILURE in 1m 39s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/3639/ : SUCCESS in 18s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/2483/ : SUCCESS in 15m 00s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/13760/ : SUCCESS in 12m 27s

**Pull request merged**